Userland/Libraries/LibCompress/Zlib.h

```diff
@@ -13,6 +13,37 @@
 
 namespace Compress {
 
+enum class ZlibCompressionMethod : u8 {
+    Deflate = 8,
+};
+
+enum class ZlibCompressionLevel : u8 {
+    Fastest,
+    Fast,
+    Default,
+    Best,
+};
+
+struct ZlibHeader {
+    union {
+        struct {
+            ZlibCompressionMethod compression_method : 4;
+            u8 compression_info : 4;
+        };
+        u8 method_and_flags;
+    };
+
+    union {
+        struct {
+            u8 check_bits : 5;
+            bool present_dictionary : 1;
+            ZlibCompressionLevel compression_level : 2;
+        };
+        u8 flags;
+    };
+};
```

Comment:
Is there a particular reason why this is not just one big union, especially since the first byte is called `method_and_flags`, and the second is called `flags`?

Reply:
Well, it’s a direct translation of what the spec says, but I will implement it better. :)